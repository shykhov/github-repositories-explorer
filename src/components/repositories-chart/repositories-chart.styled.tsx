import styled from 'styled-components';
import { Paper } from '@material-ui/core';

export const ChartWrapper = styled.div`
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  display: flex;
  justify-content: center;
`;

export const StyledPaper = styled(Paper)`
  width: 1250px;
  max-width: 100%;
  margin: 50px;
  overflow: hidden;

  @media (max-width: 1200px) {
    min-width: 800px;
    margin: 50px;
  }
`;
