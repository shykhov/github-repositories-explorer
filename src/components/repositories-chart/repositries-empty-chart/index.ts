export * from './repositries-empty-chart';
