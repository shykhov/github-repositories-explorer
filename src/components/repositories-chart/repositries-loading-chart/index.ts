export * from './repositries-loading-chart';
