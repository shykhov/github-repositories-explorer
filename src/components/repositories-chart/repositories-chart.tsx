import React, { FC } from 'react';
import { LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer } from 'recharts';
import isNil from 'lodash/isNil';
import isEmpty from 'lodash/isEmpty';

import { ContentRenderer } from '../content-renderer';
import { RepositoriesLoadingChart } from './repositries-loading-chart';
import { RepositoriesEmptyChart } from './repositries-empty-chart';
import { StyledPaper, ChartWrapper } from './repositories-chart.styled';
import { ChartDataResult } from '../../utils';

interface RepositoriesChartProps {
  chartData: ChartDataResult[];
  loading: boolean;
  repositoryNameSearchParameter: string | null;
  userLoginParameter: string | null;
}

export const RepositoriesChart: FC<RepositoriesChartProps> = props => {
  const { chartData, loading, repositoryNameSearchParameter, userLoginParameter } = props;

  const hasNoQueryParameters: boolean = isNil(userLoginParameter) && isNil(repositoryNameSearchParameter);
  const isEmptyRepositoriesList: boolean = isEmpty(chartData);

  return (
    <ChartWrapper>
      <StyledPaper elevation={20}>
        <ContentRenderer
          isLoading={loading}
          isEmpty={hasNoQueryParameters || isEmptyRepositoriesList}
          emptyComponent={<RepositoriesEmptyChart />}
          loadingComponent={<RepositoriesLoadingChart />}
          contentComponent={
            <ResponsiveContainer height={492}>
              <LineChart
                data={chartData}
                margin={{
                  top: 50,
                  right: 30,
                  left: 30,
                  bottom: 50,
                }}
              >
                <CartesianGrid strokeDasharray="3 3" />
                <XAxis hide dataKey="name" />
                <YAxis yAxisId="left" />
                <YAxis yAxisId="right" orientation="right" />
                <Tooltip />
                <Line
                  yAxisId="left"
                  isAnimationActive
                  type="monotone"
                  dataKey="stars"
                  stroke="#8884d8"
                  activeDot={{ r: 8 }}
                />
                <Line yAxisId="right" isAnimationActive type="monotone" dataKey="forks" stroke="#82ca9d" />
              </LineChart>
            </ResponsiveContainer>
          }
        />
      </StyledPaper>
    </ChartWrapper>
  );
};
