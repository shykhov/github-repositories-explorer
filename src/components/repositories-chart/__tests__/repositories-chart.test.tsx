import React from 'react';
import { render, shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';

import { RepositoriesChart, RepositoriesChartProps } from '../repositories-chart';

describe('<RepositoriesChart />', () => {
  let propsMock: RepositoriesChartProps;

  beforeEach(() => {
    propsMock = {
      loading: false,
      repositoryNameSearchParameter: 'test-repository-name',
      userLoginParameter: 'test-login-name',
      chartData: [
        {
          stars: 0,
          name: 'ASDQWEA1/ASDQWEA1',
          forks: 0,
        },
        {
          stars: 0,
          name: 'JFHWYFHEDY9FR7923RYBHFKDSCF/asdqweasdwaeasd',
          forks: 0,
        },
        {
          stars: 0,
          name: 'SLCDevelopment/asdqweasdqweaaa',
          forks: 0,
        },
      ],
    };
  });

  const renderedComponent: unknown = render(<RepositoriesChart {...propsMock} />);
  const shallowRenderedComponent: ShallowWrapper = shallow(<RepositoriesChart {...propsMock} />);

  it('should render RepositoriesChart', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should shallow render RepositoriesChart', () => {
    expect(shallowRenderedComponent).toMatchSnapshot();
  });
});
