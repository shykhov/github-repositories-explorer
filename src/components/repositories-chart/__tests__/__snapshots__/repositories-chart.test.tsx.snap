// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoriesChart /> should render RepositoriesChart 1`] = `
<div
  class="sc-AxirZ krZEiH"
>
  <div
    class="MuiPaper-root sc-AxiKw dAhZML MuiPaper-elevation20 MuiPaper-rounded"
  >
    <div
      class="sc-AxjAm ccOLpk"
    >
      <span
        aria-label="emodji"
        role="img"
      >
        ðŸ“‰
      </span>
      <div>
        Repositories informational chart will appear here.
      </div>
    </div>
  </div>
</div>
`;

exports[`<RepositoriesChart /> should shallow render RepositoriesChart 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(Paper)))
    elevation={20}
  >
    <ContentRenderer
      contentComponent={
        <ResponsiveContainer
          debounce={0}
          height={492}
          width="100%"
        >
          <LineChart
            barCategoryGap="10%"
            barGap={4}
            layout="horizontal"
            margin={
              Object {
                "bottom": 50,
                "left": 30,
                "right": 30,
                "top": 50,
              }
            }
            reverseStackOrder={false}
            stackOffset="none"
          >
            <CartesianGrid
              fill="none"
              horizontal={true}
              horizontalFill={Array []}
              horizontalPoints={Array []}
              stroke="#ccc"
              strokeDasharray="3 3"
              vertical={true}
              verticalFill={Array []}
              verticalPoints={Array []}
            />
            <XAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              dataKey="name"
              domain={
                Array [
                  0,
                  "auto",
                ]
              }
              height={30}
              hide={true}
              mirror={false}
              orientation="bottom"
              padding={
                Object {
                  "left": 0,
                  "right": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="category"
              width={0}
              xAxisId={0}
            />
            <YAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              domain={
                Array [
                  0,
                  "auto",
                ]
              }
              height={0}
              hide={false}
              mirror={false}
              orientation="left"
              padding={
                Object {
                  "bottom": 0,
                  "top": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="number"
              width={60}
              yAxisId="left"
            />
            <YAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              domain={
                Array [
                  0,
                  "auto",
                ]
              }
              height={0}
              hide={false}
              mirror={false}
              orientation="right"
              padding={
                Object {
                  "bottom": 0,
                  "top": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="number"
              width={60}
              yAxisId="right"
            />
            <Tooltip
              active={false}
              allowEscapeViewBox={
                Object {
                  "x": false,
                  "y": false,
                }
              }
              animationDuration={400}
              animationEasing="ease"
              contentStyle={Object {}}
              coordinate={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              cursor={true}
              cursorStyle={Object {}}
              filterNull={true}
              isAnimationActive={true}
              itemStyle={Object {}}
              labelStyle={Object {}}
              offset={10}
              separator=" : "
              useTranslate3d={false}
              viewBox={
                Object {
                  "x1": 0,
                  "x2": 0,
                  "y1": 0,
                  "y2": 0,
                }
              }
              wrapperStyle={Object {}}
            />
            <Line
              activeDot={
                Object {
                  "r": 8,
                }
              }
              animateNewValues={true}
              animationBegin={0}
              animationDuration={1500}
              animationEasing="ease"
              connectNulls={false}
              dataKey="stars"
              dot={true}
              fill="#fff"
              hide={false}
              isAnimationActive={true}
              legendType="line"
              onAnimationEnd={[Function]}
              onAnimationStart={[Function]}
              points={Array []}
              stroke="#8884d8"
              strokeWidth={1}
              type="monotone"
              xAxisId={0}
              yAxisId="left"
            />
            <Line
              activeDot={true}
              animateNewValues={true}
              animationBegin={0}
              animationDuration={1500}
              animationEasing="ease"
              connectNulls={false}
              dataKey="forks"
              dot={true}
              fill="#fff"
              hide={false}
              isAnimationActive={true}
              legendType="line"
              onAnimationEnd={[Function]}
              onAnimationStart={[Function]}
              points={Array []}
              stroke="#82ca9d"
              strokeWidth={1}
              type="monotone"
              xAxisId={0}
              yAxisId="right"
            />
          </LineChart>
        </ResponsiveContainer>
      }
      emptyComponent={<RepositoriesEmptyChart />}
      errorComponent={null}
      hasError={false}
      isEmpty={true}
      isLoading={false}
      loadingComponent={<RepositoriesLoadingChart />}
    />
  </Styled(WithStyles(ForwardRef(Paper)))>
</styled.div>
`;
